From f9fc70664fc6eae4351e948a89c8b3f9132dd98f Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Wed, 22 Dec 2021 06:22:34 +0200
Subject: [PATCH] [SW-67868] gaudi2: sec: bugfix for spi flash lock

Fix the end address to lock the spi flash in sram sf access function.

Change-Id: I89c49b7e582d5aa9b4f2fe10b977eea86fdfee55
---

diff --git a/arch/gaudi2/common/security.c b/arch/gaudi2/common/security.c
index 27f69f7..eb4279e 100644
--- a/arch/gaudi2/common/security.c
+++ b/arch/gaudi2/common/security.c
@@ -878,7 +878,7 @@
 		/*
 		 * Activate SPI Flash static protection.
 		 */
-		hl_secure_hbw_rr(HL_SEC_HBW_SPI_FLASH, SPI_FLASH_BASE, SPI_FLASH_BASE, rr_sel);
+		hl_secure_hbw_rr(HL_SEC_HBW_SPI_FLASH, SPI_FLASH_BASE, SPI_FLASH_END, rr_sel);
 		/*
 		 * Disable SPI Flash Read-only configuration.
 		 */
